### PR TITLE
Code clean-ups, performance tuning and documentation improvements

### PR DESCRIPTION
**Pull request by nbishdev**
As per your requests, my changes are split into smaller patches for easier review.

Let me know what you thing about them.

### CODE REVIEW

**Comment by nbishdev** [comment]
Please, review my patches so that I can move on with duplicate detection.

File: README.md [hunk 1]
```diff
@@ -73,14 +73,14 @@ To create 4-bit hashes for input data of 7 dimensions:
...
-    for ix in range(X.shape[0]):
-        x = X.getrow(ix)
-        c = y[ix]
-        lsh.index( x, extra_data=c)
+    # If you're using NumPy < v2.1.0, this indexing
+    # procedure is compatible but much slower
+    #for ix in range(X.shape[0]):
+    #    x = X.getrow(ix)
+    #    c = y[ix]
+    #    lsh.index(x, extra_data=c)
```
Comment (brandonrobertz) [inline README.md:83]: You changed the comment, making it incorrect. Please revert this change.

File: README.md [hunk 2]
```diff
@@ -177,16 +177,24 @@ Parameters:
...
 
-    lsh.index(input_point, extra_data=None)
+    # If you're using NumPy >= v2.1.0, this is much faster
+    lsh.index(X, extra_data=y)
+
+    # For all NumPy versions, but much slower
+    for ix in range(X.shape[0]):
+        x = X.getrow(ix)
+        c = y[ix]
+        lsh.index(x, extra_data=c)
```
Comment (brandonrobertz) [inline README.md:187]: Same as above

**Review by brandonrobertz — CHANGES_REQUESTED** [review]
Syntax changes are good. Please revert the rest as noted. I'll change the index documentation myself once I release the next version of the lib.

File: README.md (hunk 1 above)
Comment (nbishdev) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (nbishdev) [reply README.md]: Done.

File: README.md (hunk 1 above)
Comment (brandonrobertz) [reply README.md]: Can you revert this back to original? As it is in main currently. There's no such thing as Numpy v2

File: README.md (hunk 2 above)
Comment (brandonrobertz) [reply README.md]: See above

File: README.md (hunk 1 above)
Comment (nbishdev) [reply README.md]: Done. I though that by "using >= v2.1.0,", you meant NumPy >= v2.1.0.
I fixed it so that it is clear that the version of SparseLSH is specified.

**Comment by brandonrobertz** [comment]
can you squash this PR into a single commit?
